### PR TITLE
Rock Paper Scissors Challenge

### PR DESCRIPTION
**Pull request by DeathRay1977**

### CODE REVIEW

File: spec/game_spec.rb [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+
+    it 'delegates setting of hands to colaborators' do
+      expect(human).to receive(:give_hand)
+      expect(robot).to receive(:give_hand)
+      game.player1_hand = :rock
+      game.player2_hand = :paper
+    end
+  end
+  context 'Getting players hands' do
+    it {is_expected.to respond_to(:player1_hand)}
```
Comment (yevdyko) [inline spec/game_spec.rb]: Vacuous tests

File: spec/game_spec.rb [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+    it 'delegates getting of hands to collaborators' do
+      allow(human).to receive(:hand).and_return(:rock)
+      allow(robot).to receive(:hand).and_return(:rock)
+      expect(game.player1_hand).to eq :rock
+      expect(game.player2_hand).to eq :rock
+    end
+  end
+  context 'Play game' do
+
+    it {is_expected.to respond_to(:play)}
```
Comment (yevdyko) [inline spec/game_spec.rb]: Vacuous test

File: spec/player_spec.rb [hunk 3]
```diff
@@ -0,0 +1,25 @@
+require 'player'
+
+describe Player do
+  let(:name) {"Norm"}
+  subject(:player) {described_class.new(name)}
+
+  context 'Initialization' do
+    it {is_expected.to respond_to(:name)}
```
Comment (yevdyko) [inline spec/player_spec.rb]: Vacuous tests

File: spec/player_spec.rb [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+
+    it 'Has a name' do
+      expect(player.name).to eq name
+    end
+  end
+
+  context 'Storing the players hand' do
+    it {is_expected.to respond_to(:give_hand).with(1).argument}
```
Comment (yevdyko) [inline spec/player_spec.rb]: Vacuous tests

File: spec/computer_spec.rb [hunk 5]
```diff
@@ -0,0 +1,26 @@
+describe Computer do
+  let(:name) {"Computer"}
+  let(:rules) {double :rules, hands: [:choice1, :choice2, :choice3]}
+  let(:rules_klass) {double :rules_klass, new: rules}
+  subject(:computer) {described_class.new(rules_klass)}
+
+  context 'Initialization' do
+    it {is_expected.to respond_to(:name)}
```
Comment (yevdyko) [inline spec/computer_spec.rb]: Vacuous test

File: spec/computer_spec.rb [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+
+    it 'Has a name' do
+      expect(computer.name).to eq name
+    end
+  end
+
+  context 'Storing the players hand' do
+    it {is_expected.to respond_to(:give_hand).with(1).argument}
```
Comment (yevdyko) [inline spec/computer_spec.rb]: Vacuous tests

File: spec/game_spec.rb [hunk 7]
```diff
@@ -0,0 +1,64 @@
...
+  let(:rules) {double :rules}
+  let(:rules_klass) {double :rules_klass, new: rules}
+  subject(:game) {described_class.new(rules_klass, human, robot)}
+
+  context 'initialisation' do
+    before(:each) do
+      allow(human).to receive(:name).and_return("Norm")
+      allow(robot).to receive(:name).and_return("Computer")
+    end
+    it {is_expected.to respond_to(:player1_name)}
```
Comment (yevdyko) [inline spec/game_spec.rb]: Vacuous tests

File: spec/game_spec.rb [hunk 8]
```diff
@@ -0,0 +1,64 @@
...
+      expect(game.player1_name).to eq "Norm"
+    end
+
+    it 'is initialised with a player 2' do
+      expect(game.player2_name).to eq "Computer"
+    end
+  end
+
+  context 'Setting players hands' do
+    it {is_expected.to respond_to(:player1_hand=).with(1).argument}
```
Comment (yevdyko) [inline spec/game_spec.rb]: Vacuous tests

**Comment by tansaku** [comment]
You had your rps_challenge reviewed by **rob youngman**.
### The good points are:
* The README has instructions, The README has a screenshot(s)
* Every possible outcome of the game logic is tested at the unit level
* Win, lose and draw outcomes are all tested at the feature level
* Presentation strings not in business layer
* long if/elsif conditionals avoided in business logic
* Business and presentation logic has been cleanly separated
* No global variables used
* weapons are sensibly stored in a constant
* Routes consistently named, Routes use of POST and GET is consistent
* The code defines weapons in a single place
* The file structure followed recognizable conventions
* files and classes are named consistently
* Capybara is initialized correctly
* There is no commented code
* Feature tests stub random behaviour
* The code minimizes the amout of business logic in the view
* The controllers are thin with minimal business logic

### You should consider the following improvements:
* Testing too much game logic in feature tests

### Additional comments:
Excellent submission. Only item picked up in the review were a few vacuous tests.

**Pull request closed without merging**